storagemarket/lp2pimpl/deal_stream.go

```diff
@@ -0,0 +1,53 @@
+package lp2pimpl
+
+import (
+	"bufio"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/mux"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	cborutil "github.com/filecoin-project/go-cbor-util"
+)
+
+// DealStream reads and writes deal proposals to a libp2p stream
+type DealStream struct {
```

Comment:
This struct can be breaken down into two structs, one for the client and one for the provider.

Reply:
The client opens a deal stream -> writes a proposal -> reads a response. 

The provider simply handles incoming deal streams, reads the proposal and writes the response.